### ISSUE TITLE
Pact-broker will not start on Kubernetes with Google Cloud

### ISSUE THREAD

**Issue by steam0** [DiUS/pact_broker-docker#57]
I am trying to start the docker image on Google Cloud using Kubernetes.

I am however enountering a problem where runit cannot start new services because of a missing file.

```
[pact-broker] *** Running /etc/my_init.d/00_regen_ssh_host_keys.sh...
[pact-broker] *** Running /etc/my_init.d/30_presetup_nginx.sh...
[pact-broker] *** Running /etc/rc.local...
[pact-broker] *** Booting runit daemon...
[pact-broker] *** Runit started as PID 10
[pact-broker] warning: /etc/service/nginx-log-forwarder: unable to open supervise/ok: file does not exist
[pact-broker] ok: run: /etc/service/nginx-log-forwarder: (pid 29) 0s
```

The line in questions is `[pact-broker] warning: /etc/service/nginx-log-forwarder: unable to open supervise/ok: file does not exist`. 

This error does not happen if I try to launch the application locally on my computer.

Here is my demployment file for k8s.
```
apiVersion: apps/v1beta1 # for versions before 1.6.0 use extensions/v1beta1
kind: Deployment
metadata:
  name: pact-broker
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: pact-broker
    spec:
      containers:
      - name: pact-broker
        image: dius/pact-broker:latest
        imagePullPolicy: Always
        ports:
        - name: api
          containerPort: 80
          protocol: TCP
        resources:
          limits:
            memory: 512Mi
            cpu: 100m
          requests:
            memory: 256Mi
            cpu: 50m
        envFrom:
          - configMapRef:
              name: pact-broker-config
        env:
          - name: PACT_BROKER_DATABASE_HOST
            value: "127.0.0.1:5432"
          - name: PACT_BROKER_DATABASE_NAME
            value: "pactbroker"
          - name: PACT_BROKER_DATABASE_USERNAME
            valueFrom:
              secretKeyRef:
                name: pact-broker-db-credentials
                key: username
          - name: PACT_BROKER_DATABASE_PASSWORD
            valueFrom:
              secretKeyRef:
                name: pact-broker-db-credentials
                key: password
      - image: gcr.io/cloudsql-docker/gce-proxy:1.11
        name: cloudsql-proxy
        command: ["/cloud_sql_proxy", "--dir=/cloudsql",
                  "-instances=entur-1287:europe-west1:pact-broker-db-dev=tcp:5432",
                  "-credential_file=/secrets/cloudsql/credentials.json"]
        volumeMounts:
          - name: pact-broker-instance-credentials
            mountPath: /secrets/cloudsql
            readOnly: true
          - name: ssl-certs
            mountPath: /etc/ssl/certs
          - name: cloudsql
            mountPath: /cloudsql
      volumes:
        - name: pact-broker-instance-credentials
          secret:
            secretName: pact-broker-instance-credentials
        - name: ssl-certs
          hostPath:
            path: /etc/ssl/certs
        - name: cloudsql
          emptyDir:
```

**TL;DR**
Seems like the docker image behaves differently on Kubernetes than on a local computer.

**Comment by mefellows** [comment]
Hi @steam0, I can't see anything immediately wrong with your configuration, however I will say that it definitely works on Kube as we use Kube to run our hosted [Pact Broker service](http://pact.dius.com.au). 

I'm guessing this is on GCE?

That looks like a non-fatal error, does the app still start? Are you able to provide the entire log file too?

**Comment by steam0** [comment]
I don't know why it would not start, but as of now it will _not_ start on first apply. But when I upgrade my configuration and pods are restarted, it starts just fine.

**Comment by bethesque** [comment]
Any further thoughts on this? I'll close it unless we have an outstanding action. We can always reopen if need be.

**Comment by steam0** [comment]
Something is definitely wrong with the image and how I am running it on GCE. But since I don't know what, then let's just close this off for now. Maybe someone else finds it later and have a solution.

**Comment by bethesque** [comment]
Ok, fingers crossed!
